src/AspNetCoreModuleV2/AspNetCore/src/aspnetcore_shim_config.cpp

```diff
@@ -0,0 +1,260 @@
+            pAspNetCoreShimConfig->QueryProcessPath()->QueryStr(),
+            pAspNetCoreShimConfig->QueryApplicationPhysicalPath()->QueryStr(),
+            pAspNetCoreShimConfig->QueryArguments()->QueryStr(),
+            &struHostFxrDllLocation,
+            &dwArgCount,
+            &pwzArgv)))
+        {
+            goto Finished;
+        }
+
+        if (FAILED(hr = pAspNetCoreShimConfig->SetHostFxrFullPath(struHostFxrDllLocation.QueryStr())))
+        {
+            goto Finished;
+        }
+
+        pAspNetCoreShimConfig->SetHostFxrArguments(dwArgCount, pwzArgv);
+    }
+
+    hr = pHttpApplication->GetModuleContextContainer()->
+        SetModuleContext(pAspNetCoreShimConfig, pModuleId);
+    if (FAILED(hr))
+    {
+        if (hr == HRESULT_FROM_WIN32(ERROR_ALREADY_ASSIGNED))
+        {
+            delete pAspNetCoreShimConfig;
+
+            pAspNetCoreShimConfig = (ASPNETCORE_SHIM_CONFIG*)pHttpApplication->
+                GetModuleContextContainer()->
+                GetModuleContext(pModuleId);
+
+            _ASSERT(pAspNetCoreShimConfig != NULL);
+
+            hr = S_OK;
+        }
+        else
+        {
+            goto Finished;
+        }
+    }
+    else
+    {
+        DebugPrintf(ASPNETCORE_DEBUG_FLAG_INFO,
+            "ASPNETCORE_SHIM_CONFIG::GetConfig, set config to ModuleContext");
+
+        // set appliction info here instead of inside Populate()
+        // as the destructor will delete the backend process
+        hr = pAspNetCoreShimConfig->QueryApplicationPath()->Copy(pHttpApplication->GetApplicationId());
+        if (FAILED(hr))
+        {
+            goto Finished;
+        }
+    }
+
+    *ppAspNetCoreShimConfig = pAspNetCoreShimConfig;
+    pAspNetCoreShimConfig = NULL;
+
+Finished:
+
+    if (pAspNetCoreShimConfig != NULL)
+    {
+        delete pAspNetCoreShimConfig;
+        pAspNetCoreShimConfig = NULL;
+    }
+
+    return hr;
+}
+
+
+HRESULT
+ASPNETCORE_SHIM_CONFIG::Populate(
+    IHttpServer    *pHttpServer,
+    IHttpContext   *pHttpContext
+)
+{
+    STACK_STRU(strHostingModel, 300);
+    HRESULT                         hr = S_OK;
+    STRU                            strApplicationFullPath;
+    IAppHostAdminManager           *pAdminManager = NULL;
+    IAppHostElement                *pAspNetCoreElement = NULL;
+    DWORD                           dwCounter = 0;
+    DWORD                           dwPosition = 0;
+    WCHAR*                          pszPath = NULL;
+    BSTR                            bstrAspNetCoreSection = NULL;
+
+    pAdminManager = pHttpServer->GetAdminManager();
+    hr = m_struConfigPath.Copy(pHttpContext->GetApplication()->GetAppConfigPath());
+    if (FAILED(hr))
+    {
+        goto Finished;
+    }
+
+    hr = m_struApplicationPhysicalPath.Copy(pHttpContext->GetApplication()->GetApplicationPhysicalPath());
+    if (FAILED(hr))
+    {
+        goto Finished;
+    }
+
+    pszPath = m_struConfigPath.QueryStr();
+    while (pszPath[dwPosition] != NULL)
```

Comment:
What is this?

Reply:
apppath (pszPath) is not used by shim-config and can be removed.